File: src/main/java/org/chousik/collection/builder/collectors/LongCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class LongCollector extends NumberCollector<Long>{

    @Override
    public Long ask(String name, IValidator<Long> validator) throws ScriptExecutionError {
        return askNumber(name, validator, Long::parseLong);
    }
}

File: src/main/kotlin/collection/builder/collectors/LongCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.validators.IValidator

class LongCollector : NumberCollector<Long>() {
    override fun ask(name: String, validator: IValidator<Long?>): Long {
        return askNumber(name, validator, java.lang.Long::parseLong)
    }
}